modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/components/SubscriptionPlanRow.kt

```diff
@@ -0,0 +1,327 @@
+                        stringResource(LR.string.price_per_month_usd, plan.pricePerMonth)
+                    } else {
+                        stringResource(LR.string.price_per_month, plan.pricePerMonth, currencyCode)
+                    },
+                    color = MaterialTheme.theme.colors.primaryText02,
+                    fontSize = 15.sp,
+                    lineHeight = 21.sp,
+                    modifier = Modifier.align(Alignment.CenterVertically),
+                )
+            }
+        }
+
+        if (plan.tier == SubscriptionTier.Plus && plan.billingCycle == BillingCycle.Yearly) {
+            Box(
+                modifier = Modifier
+                    .align(
+                        when (badgeConfig.badgePosition) {
+                            BadgePosition.RIGHT -> Alignment.TopEnd
+                            BadgePosition.CENTER -> Alignment.TopCenter
+                        },
+                    )
+                    .offset(x = (-10).dp, y = (-10).dp)
+                    .background(MaterialTheme.theme.colors.primaryField03Active, CircleShape)
+                    .padding(horizontal = 12.dp, vertical = 2.dp),
+            ) {
+                TextH50(
+                    text = badgeConfig.badgeText.text(),
+                    color = MaterialTheme.theme.colors.primaryUi01,
+                    disableAutoScale = true,
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun CheckMark(
+    isSelected: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    if (isSelected) {
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier = modifier.background(
+                color = MaterialTheme.theme.colors.primaryField03Active,
+                shape = CircleShape,
+            ),
+        ) {
+            Image(
+                painter = painterResource(IR.drawable.ic_checkmark_small),
+                colorFilter = ColorFilter.tint(MaterialTheme.theme.colors.primaryUi01),
+                contentDescription = null,
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(5.dp),
+            )
+        }
+    } else {
+        Box(
+            modifier = modifier.border(
+                width = 1.dp,
+                color = MaterialTheme.theme.colors.primaryIcon02,
+                shape = CircleShape,
+            ),
+        )
+    }
+}
+
+private val SubscriptionPlan.Base.pricePerMonth: Float
+    get() {
+        val pricePerWeek = when (billingCycle) {
+            BillingCycle.Monthly -> pricingPhase.price.amount
+            BillingCycle.Yearly -> pricingPhase.price.amount / 12.toBigDecimal()
+        }
+        return pricePerWeek.toFloat()
+    }
+
+private fun SubscriptionPlan.Base.savingsPercent(otherPlan: SubscriptionPlan.Base): Int {
+    if (this.billingCycle != BillingCycle.Yearly || otherPlan.billingCycle != BillingCycle.Monthly) return 0
```

Comment:
Why is this condition necessary? Why can't any configuration of billing cycles be compared?